xcode/Subconscious/Shared/Components/AddressBook/AddFriendView.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  AddFriendView.swift
+//  Subconscious (iOS)
+//
+//  Created by Ben Follington on 2/27/23.
+//
+
+import SwiftUI
+import ObservableStore
+
+struct AddFriendView: View {
+    var state: AddressBookModel
+    var send: (AddressBookAction) -> Void
+    
+    @Environment(\.presentationMode) var presentationMode: Binding<PresentationMode>
+    
+    @State var did: String = ""
+    @State var petname: String = ""
+    
+    func validateDid(key: String) -> Did? {
+        Did(key)
+    }
+    
+    func validatePetname(petname: String) -> Petname? {
+        Petname(petname)
+    }
+    
+    var body: some View {
+        NavigationStack {
+            Form {
+                Section(header: Text("Friend Details")) {
+                    HStack(alignment: .top) {
+                        Image(systemName: "key")
+                            .foregroundColor(.accentColor)
+                        ValidatedTextField(
+                            placeholder: "DID",
+                            text: $did,
+                            caption: "e.g. did:key:z6MkmCJAZansQ3p1Qwx6wrF4c64yt2rcM8wMrH5Rh7DGb2K7",
+                            isValid: validateDid(key: did) != nil || did.count  == 0 // Prevent initial error
```

Comment:
I came up with a slightly nicer version of this in https://github.com/subconsciousnetwork/subconscious/pull/429